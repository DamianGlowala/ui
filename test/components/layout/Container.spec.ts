import { describe, it, expect } from 'vitest'
import { UContainer } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Container', () => {
  it.each([
    ['basic case', {}],
    ['with as', { as: 'article' }],
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof Container.props>) => {
    if (options !== undefined) {
      options.slots = options.slots || { default: () => 'label' }
      options.slots.default = options.slots.default || (() => 'label')
    }
    const html = await ComponentRender(nameOrHtml, options, UContainer)
    expect(html).toMatchSnapshot()
  })
})
