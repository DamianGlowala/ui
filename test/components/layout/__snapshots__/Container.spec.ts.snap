// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container > renders basic case correctly 1`] = `"<div class="mx-auto px-4 sm:px-6 lg:px-8 max-w-7xl">label</div>"`;

exports[`Container > renders wit as correctly 1`] = `"<div class="mx-auto px-4 sm:px-6 lg:px-8 max-w-7xl">label</div>"`;

exports[`Container > renders with as correctly 1`] = `"<div class="mx-auto px-4 sm:px-6 lg:px-8 max-w-7xl">label</div>"`;
